shared/src/main/scala/mlscript/TypeSimplifier.scala

```diff
@@ -385,13 +388,14 @@ trait TypeSimplifier { self: Typer =>
                         )
                       val componentFieldsMap = componentFields.toMap
                       val tupleComponents = fs.iterator.zipWithIndex.map { case ((nme, ty), i) =>
-                        nme -> go(ty & componentFieldsMap.getOrElse(i + 1, TopType), pol)
+                        nme -> (ty && componentFieldsMap.getOrElse(i + 1, TopType.toUpper(noProv))).update(go(_, !pol), go(_, pol))
                       }.toList
-                      S(TupleType(tupleComponents)(tt.prov)) -> rcdFields.mapValues(go(_, pol))
+                      S(TupleType(tupleComponents)(tt.prov)) -> rcdFields.mapValues(_.update(go(_, !pol), go(_, pol)))
                     case S(ct: ClassTag) => S(ct) -> nFields
-                    case S(ft @ FunctionType(l, r)) =>
+                    case S(ft @ FunctionType(l, r)) => 
```

Comment:
```suggestion
                    case S(ft @ FunctionType(l, r)) =>
```

Reply:
IS this resolved? If so, please mark it so.

Reply:
yep

Reply:
> If so, please mark it so. (resolved)